apps/rush-lib/src/logic/base/BaseInstallManager.ts

```diff
@@ -0,0 +1,652 @@
+
+    // This marker file indicates that the last "rush install" completed successfully.
+    // If "--purge" was specified, or if the last install was interrupted, then we will need to
+    // perform a clean install.  Otherwise, we can do an incremental install.
+    const cleanInstall: boolean = !this._commonTempInstallFlag.checkValidAndReportStoreIssues();
+
+    // Allow us to defer the file read until we need it
+    const canSkipInstall: () => boolean = () => {
+      // Based on timestamps, can we skip this install entirely?
+      const outputStats: fs.Stats = FileSystem.getStatistics(this._commonTempInstallFlag.path);
+      return this.canSkipInstall(outputStats.mtime);
+    };
+
+    if (cleanInstall || !shrinkwrapIsUpToDate || !variantIsUpToDate || !canSkipInstall()) {
+      let publishedRelease: boolean | undefined;
+      try {
+        publishedRelease = await this._checkIfReleaseIsPublished();
+      } catch {
+        // If the user is working in an environment that can't reach the registry,
+        // don't bother them with errors.
+      }
+
+      if (publishedRelease === false) {
+        console.log(
+          colors.yellow('Warning: This release of the Rush tool was unpublished; it may be unstable.')
+        );
+      }
+
+      // Since we're going to be tampering with common/node_modules, delete the "rush link" flag file if it exists;
+      // this ensures that a full "rush link" is required next time
+      Utilities.deleteFile(this.rushConfiguration.rushLinkJsonFilename);
+
+      // Delete the successful install file to indicate the install transaction has started
+      this._commonTempInstallFlag.clear();
+
+      // Perform the actual install
+      await this.install(cleanInstall);
+
+      const usePnpmFrozenLockfile: boolean =
+        this._rushConfiguration.packageManager === 'pnpm' &&
+        this._rushConfiguration.experimentsConfiguration.configuration.usePnpmFrozenLockfileForRushInstall ===
+          true;
+
+      if (this.options.allowShrinkwrapUpdates && (usePnpmFrozenLockfile || !shrinkwrapIsUpToDate)) {
+        // Shrinkwrap files may need to be post processed after install, so load and save it
+        const tempShrinkwrapFile: BaseShrinkwrapFile | undefined = ShrinkwrapFileFactory.getShrinkwrapFile(
+          this._rushConfiguration.packageManager,
+          this._rushConfiguration.packageManagerOptions,
+          this._rushConfiguration.tempShrinkwrapFilename
+        );
+        if (tempShrinkwrapFile) {
+          tempShrinkwrapFile.save(this._rushConfiguration.tempShrinkwrapFilename);
+        }
+
+        // Copy (or delete) common\temp\pnpm-lock.yaml --> common\config\rush\pnpm-lock.yaml
+        Utilities.syncFile(
+          this._rushConfiguration.tempShrinkwrapFilename,
+          this._rushConfiguration.getCommittedShrinkwrapFilename(this.options.variant)
+        );
+      } else {
+        // TODO: Validate whether the package manager updated it in a nontrivial way
```

Comment:
This has been a todo for quite some time... I think it's more of a "for future considerations" type of todo. Though it can be removed if you think it mucks up the file

Reply:
Eh just go ahead and leave it then.